### PR TITLE
sci-libs/scikit-learn: version bump to 0.24.1

### PR DESCRIPTION
**Pull request by epsilon-0**
Package-Manager: Portage-3.0.12, Repoman-3.0.2
Signed-off-by: Aisha Tammy <gentoo@aisha.cc>

---

~~tests are not working yet~~
Manually fixed tests.

### CODE REVIEW

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		      "${tfile}" =~ "test_pipeline.py" ]]; then
+			continue
+		fi
+		elog "Testing: ${tfile}"
+		${EPYTHON} "${tfile}" || die "tests failed for ${tfile} with ${EPYTHON}"
+	done
+}
+
+python_install_all() {
+	find "${S}" -name \*LICENSE.txt -delete
```
Comment (SoapGentoo) [inline sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild:60]: missing `|| die`

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	distutils-r1_python_install_all
+	if use examples; then
+		dodoc -r examples
+	fi
```
Comment (SoapGentoo) [inline sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild:64]: you can do this in one line

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	# remove bundled cblas
+	rm -rf sklearn/src || die "failed to remove bundled cblas"
+	distutils-r1_python_prepare_all
+}
+
+python_test() {
+	distutils_install_for_testing
+	# manually run tests as they need some weird thingies
+	# skip tests which need data files that are not installed
+	for tfile in sklearn/tests/test_*.py ; do
```
Comment (SoapGentoo) [inline sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild:48]: define `tfile` as local before

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	dev-python/threadpoolctl[${PYTHON_USEDEP}]
+"
+
+distutils_enable_tests pytest
+
+PATCHES=( "${FILESDIR}"/${P}-no-O3.patch )
+
+python_prepare_all() {
+	# remove bundled cblas
+	rm -rf sklearn/src || die "failed to remove bundled cblas"
```
Comment (SoapGentoo) [inline sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild:40]: why the `-f`?

File: sci-libs/scikit-learn/files/scikit-learn-0.24.0-no-O3.patch [hunk 5]
```diff
@@ -0,0 +1,55 @@
+diff --git a/sklearn/manifold/setup.py b/sklearn/manifold/setup.py
+index 0db2d5d..fb4fc38 100644
+--- a/sklearn/manifold/setup.py
++++ b/sklearn/manifold/setup.py
+@@ -15,14 +15,14 @@ def configuration(parent_package="", top_path=None):
+     config.add_extension("_utils",
+                          sources=["_utils.pyx"],
+                          include_dirs=[numpy.get_include()],
+-                         libraries=libraries,
+-                         extra_compile_args=["-O3"])
```
Comment (SoapGentoo) [inline sci-libs/scikit-learn/files/scikit-learn-0.24.0-no-O3.patch:10]: can you talk to upstream abotu this?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sci-libs/scikit-learn/files/scikit-learn-0.24.0-no-O3.patch (hunk 5 above)
Comment (epsilon-0) [reply sci-libs/scikit-learn/files/scikit-learn-0.24.0-no-O3.patch:10]: I'll definitely ask them but...
honestly, this build system is so bad, it keeps changing every version.
like in last version they weren't even using these functions. I've like no clue what they are gonna do next version.
so frigging irritating.

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild (hunk 1 above)
Comment (epsilon-0) [reply sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild]: done

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild (hunk 2 above)
Comment (epsilon-0) [reply sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild:65]: done

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild (hunk 3 above)
Comment (epsilon-0) [reply sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild:49]: done

File: sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild (hunk 4 above)
Comment (epsilon-0) [reply sci-libs/scikit-learn/scikit-learn-0.24.0.ebuild]: removed -f.

**Comment by AndrewAmmerlaan** [comment]
Thanks :+1: 

I took the liberty of bumping to 0.24.2 while merging, and dropping 0.23.2 which was affected by security bug https://bugs.gentoo.org/758323 hope you don't mind.

**Comment by epsilon-0** [comment]
yep, that works with me.
thanks!

**Pull request closed without merging**